relay-server/src/actors/envelopes.rs

```diff
@@ -66,6 +66,7 @@ pub struct SendEnvelope {
     pub http_encoding: HttpEncoding,
     pub response_sender: Option<oneshot::Sender<Result<(), SendEnvelopeError>>>,
     pub project_key: ProjectKey,
+    partition_key: Option<String>,
```

Comment:
Having a `String` here means that other envelopes could theoretically use the same header with a different partitioning strategy.